pkg/controller/identity/iampolicy/controller.go

```diff
@@ -99,7 +103,14 @@ func (e *external) Observe(ctx context.Context, mgd resource.Managed) (managed.E
 	}
 
 	if meta.GetExternalName(cr) == "" {
-		return managed.ExternalObservation{}, nil
+		// If external name not set there is still a change it may already exist
+		// Try to get the policy by name
```

Comment:
any reason for not returning, or at least logging, unexpected errors here and in 
`kube.Update`?

Reply:
the main reason is - if this fails it will silently fall back to the previous behavior.

`kube.Update` - IMO it would be better to not fail if kube can't be updated - and I *think* there is already logging included within that (???)

Thinking about this from the end user perspective - what you said makes sense - logging/returning a better error makes sense.  I added a new exception message specific to the external-name lookup failures.
What end users should know when they see an exception in that block is - there is something wrong with the external-name field which they have to correct. (and if the external name field is correct then there is something they need to correct as to why to why their policy can't be observed)